module/batch-action/src/Domain/ValueObject/BatchActionType.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\BatchAction\Domain\ValueObject;
+
+class BatchActionType
+{
+    public const MAX_LENGTH = 20;
+
+    private string $value;
+
+    public function __construct(string $value)
+    {
+        $this->value = strtolower(trim($value));
+
+        if (!self::isValid($this->value)) {
+            throw new \InvalidArgumentException(\sprintf('Value "%s" is not valid batch action type', $value));
+        }
+    }
+
+    public function getValue(): string
+    {
+        return $this->value;
+    }
+
+    public function __toString(): string
+    {
+        return $this->value;
+    }
+
+    public function isEqual(self $code): bool
+    {
+        return $code->value === $this->value;
```

Comment:
value is not public property.

Reply:
php allow that different instances of the same class, can access private properties/methods of each other.

Reply:
That just broke my entire picture of things 😢